app/views/hosts/console/vnc.html.erb

```diff
@@ -9,11 +9,22 @@
           {:id => "troubleshootingButton", :class => "btn btn-default"}
         )
  ) %>
-<p class="toggle-hidden-value"><%= _('Password:') %> <a href="#" class="toggle-hidden-value"><%=icon_text("plus")%></a><span><%= @console[:password] %></span></p>
-<div id='vnc' data-port='<%= @console[:proxy_port] %>' data-password='<%= @console[:password] %>' data-encrypt='<%= @encrypt %>'>
-  <div id="noVNC_status_bar" class="console-status">
-    <div id="noVNC_status" class="col-md-12 label">Loading ...</div>
-  </div>
+<% if @console[:proxy_port] %>
+  <div id='vnc' data-port='<%= @console[:proxy_port] %>' data-password='<%= @console[:password] %>' data-encrypt='<%= @encrypt %>'>
+
+<% else %>
+  <div id='vnc'
+       data-host='<%= @console[:host] %>'
+       data-port='<%= @console[:port] %>'
+       data-path='<%= @console[:path] %>'
+       data-token_protocol='<%= @console[:token_protocol] %>'
+       data-plain_kubevirt_protocol='<%= @console[:plain_kubevirt_protocol] %>'
```

Comment:
It would be better to name it with a generic name and not tie the core code to a specific provider.